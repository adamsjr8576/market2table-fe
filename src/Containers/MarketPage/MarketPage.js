import React from 'react';
import { withRouter } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './MarketPage.scss';
import { addSelectedMarket, addVendors } from '../../actions';
import { getVendorsByMarketId } from '../../apiCalls';

export const MarketPage = ({ history }) => {
  const {selectedMarket, markets} = useSelector(state => state);
  const dispatch = useDispatch();

  const marketInfo = markets.find(market => market.id === selectedMarket);

  const handleSubmit = () => {
    history.push(`${selectedMarket}/vendors`)
    getVendorsByMarketId(selectedMarket)
    .then(vendors => {
      dispatch(addVendors(vendors.data.market.vendors));
    })
    .catch(error => console.log(error))
  }

  return (
    <section className='section-market-page-container'>
      <h2>{marketInfo.name}</h2>
      <h3>Address: </h3>
      <p>{marketInfo.address}</p>
      <h3>Schedule: </h3>
      <p>{marketInfo.schedule}</p>
      <h3>Google Maps: </h3>
      <a href={marketInfo.google_link}>Click Here For Directions</a>
      <button className='find-vendors-button'onClick={handleSubmit}>
        Find Vendors
      </button>
    </section>
  )
}

export default withRouter(MarketPage);
