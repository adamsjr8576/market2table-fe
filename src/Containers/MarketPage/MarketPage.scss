.section-market-page-container {
  width: 40%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  overflow: auto;
  background-color: white;
  border-right: 2px solid black;
  margin: auto;
  text-align: center;
}
.section-market-info-container {
  width: 90%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  overflow: auto;
}

.market-name{
  text-align: center;
}

.market-schedule{
  text-align: center;
}

.find-vendors-button{
  font-size: 1.5em;
  border-radius: 20px;
  padding: 5px;
  min-width: 100px;
  cursor: pointer;
  text-align: center;
  margin-top: 30px;
  outline: 0;
  width: 50%;
  background-color: #5E7838;
  color: #FAF6F5;
}
.find-vendors-button:hover,
.find-vendors-button:focus {
  transform: scale(1.05);
}

@media screen and (max-width: 1050px) {
  .section-market-page-container {
    width: 85%;
    margin: auto;
    height: 52%;
  }
}
