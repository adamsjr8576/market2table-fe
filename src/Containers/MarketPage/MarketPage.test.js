import React from 'react';
import { shallow } from 'enzyme';
import { MarketPage } from './MarketPage';

const mockState = {
  markets: [{name: 'Friendly Famrs', address: '1234 Hillsbury Lane', schedule: 'M-F ALL DAY', google_link: 'google.com', id:'1923782'}],
  selectedMarket: '1923782'
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState,
  useDispatch: () => jest.fn()
}));

describe('MarketPage', () => {

  let wrapper;
  let mockAddSelectedMarket = jest.fn();
  let mockAddVendors = jest.fn();
  let mockhistory = {push: jest.fn()};

  it('should match the snapshot', () => {
    wrapper = shallow(<MarketPage
      history={mockhistory}
      addSelectedMarket={mockAddSelectedMarket}
      addVendors={mockAddVendors}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
