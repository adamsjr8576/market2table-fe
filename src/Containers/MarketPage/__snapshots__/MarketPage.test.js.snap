// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketPage should match the snapshot 1`] = `
<section
  className="section-market-page-container"
>
  <div
    className="section-market-info-container"
  >
    <h2
      className="market-name"
    >
      Friendly Famrs
    </h2>
    <h3>
      Address: 
    </h3>
    <p>
      1234 Hillsbury Lane
    </p>
    <h3>
      Schedule: 
    </h3>
    <p
      className="market-schedule"
    >
      M-F ALL DAY
    </p>
    <h3>
      Google Maps: 
    </h3>
    <a
      href="google.com"
    >
      Click Here For Directions
    </a>
  </div>
  <button
    className="find-vendors-button"
    onClick={[Function]}
  >
    Find Vendors
  </button>
</section>
`;
